keps/sig-apps/3939-include-terminating-pods-as-active/README.md

```diff
@@ -0,0 +1,781 @@
+
+This case was added due to a bug discovered with running IndexedJobs with Tensorflow.  See [Jobs create replacement Pods as soon as a Pod is marked for deletion](https://github.com/kubernetes/kubernetes/issues/115844) for more details.
+
+#### Story 2
+
+As a cloud user, users would want to guarantee that the number of pods that are running is exactly the amount that they specify.  Terminating pods do not relinguish resources so scarce compute resource are still scheduled to those pods.
+See [Kueue: Account for terminating pods when doing preemption](https://github.com/kubernetes-sigs/kueue/issues/510) for an example of this.
+
+#### Story 3
+
+As a cloud user, users would want to guarantee that the number of pods that are running includes terminating pods.  In scare compute environments, users may only have a limited amount of nodes and they do not want to try and schedule pods to a new resource.
+Counting terminating pods as active allows for the scheduling of pods to wait until pods are terminated.
+
+See [Option for acknowledging terminating Pods in Deployment rolling update](https://github.com/kubernetes/kubernetes/issues/107920)
+for more examples.
+
+### Notes/Constraints/Caveats (Optional)
+
+The Deployment API is open for discussion.  We put the field in Deployment/ReplicaSet because it is related to RolloutStrategy.
+It is not clear if `recreate` and/or `rollingupdate` need this API for both rollout options.
+
+Another open question is if we want to include Deployments in the initial release of this feature.  There is some discussion about releasing the Job API first and then follow up with Deployment.  
+
+We decided to define the APIs in this KEP as they can utilize the same implementation.  
+
+### Risks and Mitigations
+
+<!--
+What are the risks of this proposal, and how do we mitigate? Think broadly.
+For example, consider both security and how this will impact the larger
+Kubernetes ecosystem.
+
+How will security be reviewed, and by whom?
+
+How will UX be reviewed, and by whom?
+
+Consider including folks who also work outside the SIG or subproject.
+-->
+
+## Design Details
+
+### Job API Definition
+
+At the JobSpec level, we are adding a new BoolPtr field:
+
+```golang
+type JobSpec struct{
+  ...
+ // terminatingAsActive specifies if the Job controller should include terminating pods
+ // as active. If the field is true, then the Job controller will include active pods
+ // to mean running or terminating pods
+ // +optional
+ TerminatingAsActive *bool
+}
+```
+
+### Deployment/ReplicaSet API
+
+```golang
+// DeploymentStrategy stores information about the strategy and rolling-update
+// behavior of a deployment.
+type DeploymentStrategy struct {
+  ... 
+  // TerminatingAsActive specifies if the Deployments should include terminating pods
+ // as active. If the field is true, then the Deployment controller will include active pods
+ // to mean running or terminating pods
+ // +optional
+ TerminatingAsActive *bool
```

Comment:
I don’t believe this is related to update strategy because it applies to replacement (not controlled by update strategy).

Reply:
Moved this up to deployment spec.